lib/charms/mongodb/v1/mongodb_tls.py

```diff
@@ -99,17 +105,21 @@ def request_certificate(
         internal: bool,
     ):
         """Request TLS certificate."""
+        if not self.charm.model.get_relation(Config.TLS.TLS_PEER_RELATION):
```

Comment:
This is fine, though overly defensive and may hide some hard to find bugs along the lines. 
We shouldn't have this workflow run unless the relation is established. I prefer we remove this check and make this method private

Reply:
unfortunately this lib is owned by the VM mongodb charm. Is this a blocker?

Reply:
[the changes you request must be made as a separate pr](https://github.com/canonical/mongodb-operator/pull/504) - only once it is merged can I update this one